static_report/src/components/Serve/ServeIndex.tsx

```diff
@@ -0,0 +1,234 @@
+  return (
+    <Center>
+      <Box sx={style}>
+        <Center>
+          <Box w={70} textAlign={['left']}>
+            <Text>Base: </Text>
+            <Text>Target: </Text>
+          </Box>
+          <Box mr={30}>
+            <Text>{checkState[0]}</Text>
+            <Text>{checkState[1]}</Text>
+          </Box>
+          <Box m={'auto 0'}>
+            <Button onClick={onCompareClick}>Compare</Button>
+          </Box>
+        </Center>
+      </Box>
+    </Center>
+  );
+}
+
+export function ServeIndex({ data }) {
+  const [checkState, setCheckState] = useState<string[]>([]);
+  const forceUpdate = useForceUpdate();
+
+  const onSelectChanged = (name: string, checked: boolean) => {
+    setCheckState((prev) => {
+      const index = prev.indexOf(name);
+      if (checked && index === -1) {
+        prev.push(name);
+        if (prev.length > 2) {
+          prev.shift();
+        }
+      } else if (!checked && index >= 0) {
+        prev.splice(index, 1);
+      }
+      return prev;
+    });
+  };
+
+  const onCompareClick = (base: string, target: string) => {
+    fetch(`/compare?base=${base}&target=${target}`)
+      .then((res) => res.json())
+      .then(() => {
+        alert("Compare report generated. Please check 'Comparisons' tab.");
+        window.location.reload();
+      });
+  };
+
+  return (
+    <Main>
+      <Flex
+        direction="column"
+        border="1px solid"
+        borderColor="gray.300"
+        bg="white"
+        borderRadius="md"
+        p={6}
+        width="calc(100% - 64px)"
+        maxWidth="1200px"
+        my={10}
+      >
+        <Heading mb={4}>PipeRider Reports</Heading>
+
+        <Tabs isLazy>
+          <TabList>
+            <Tab>Single runs</Tab>
+            <Tab>Comparisons</Tab>
+          </TabList>
+
+          <TabPanels>
+            <TabPanel>
+              <TableContainer>
+                <Table variant="simple">
+                  <Thead>
+                    <Tr>
+                      <Th />
+                      <Th>Name</Th>
+                      <Th>Data Source</Th>
+                      <Th>Generated At</Th>
+                    </Tr>
+                  </Thead>
+                  <Tbody>
+                    {data &&
+                      data.single.map((item) => (
+                        <LinkBox as={Tr} _hover={{ bg: 'blackAlpha.50' }}>
+                          <Td w={0}>
+                            <Checkbox
+                              value={item.name}
+                              size="lg"
+                              zIndex={2}
+                              isChecked={checkState.includes(item.name)}
+                              onChange={(event) => {
+                                onSelectChanged(
+                                  item.name,
+                                  event.target.checked,
+                                );
+                                forceUpdate();
```

Comment:
I think `useEffect` with dep array of `checkState` would be cleaner in this situation... 
as we need this to trigger on first-render, and subsequently on change of another state `checkState`.

Reply:
after discussed with Jon, we refined the behavior of `checkState` and rendering.